core/code/map_data_render.js

```diff
@@ -267,125 +264,101 @@ window.Render.prototype.createPlaceholderPortalEntity = function(guid,latE6,lngE
-  if (team == TEAM_NONE) portalLevel = 0;
-
   var latlng = L.latLng(data.latE6/1E6, data.lngE6/1E6);
 
-  var dataOptions = {
-    level: portalLevel,
-    team: team,
-    ent: ent,  // LEGACY - TO BE REMOVED AT SOME POINT! use .guid, .timestamp and .data instead
-    guid: ent[0],
-    timestamp: ent[1],
-    data: data
-  };
-
-  window.pushPortalGuidPositionCache(ent[0], data.latE6, data.lngE6);
-
-  var marker = createMarker(latlng, dataOptions);
-
-  function handler_portal_click (e) {
-    window.renderPortalDetails(e.target.options.guid);
-  }
-  function handler_portal_dblclick (e) {
-    window.renderPortalDetails(e.target.options.guid);
-    window.map.setView(e.target.getLatLng(), DEFAULT_ZOOM);
-  }
-  function handler_portal_contextmenu (e) {
-    window.renderPortalDetails(e.target.options.guid);
-    if (window.isSmartphone()) {
-      window.show('info');
-    } else if (!$('#scrollwrapper').is(':visible')) {
-      $('#sidebartoggle').click();
-    }
-  }
-
-  marker.on('click', handler_portal_click);
-  marker.on('dblclick', handler_portal_dblclick);
-  marker.on('contextmenu', handler_portal_contextmenu);
-
-  window.runHooks('portalAdded', {portal: marker, previousData: previousData});
-
-  window.portals[ent[0]] = marker;
+  window.pushPortalGuidPositionCache(data.guid, data.latE6, data.lngE6);
 
   // check for URL links to portal, and select it if this is the one
-  if (urlPortalLL && urlPortalLL[0] == marker.getLatLng().lat && urlPortalLL[1] == marker.getLatLng().lng) {
+  if (urlPortalLL && urlPortalLL[0] == latlng.lat && urlPortalLL[1] == latlng.lng) {
     // URL-passed portal found via pll parameter - set the guid-based parameter
-    log.log('urlPortalLL '+urlPortalLL[0]+','+urlPortalLL[1]+' matches portal GUID '+ent[0]);
+    log.log('urlPortalLL '+urlPortalLL[0]+','+urlPortalLL[1]+' matches portal GUID '+data.guid);
 
-    urlPortal = ent[0];
+    urlPortal = data.guid;
     urlPortalLL = undefined;  // clear the URL parameter so it's not matched again
   }
-  if (urlPortal == ent[0]) {
+  if (urlPortal == data.guid) {
     // URL-passed portal found via guid parameter - set it as the selected portal
     log.log('urlPortal GUID '+urlPortal+' found - selecting...');
-    selectedPortal = ent[0];
+    selectedPortal = data.guid;
     urlPortal = undefined;  // clear the URL parameter so it's not matched again
   }
 
-  // (re-)select the portal, to refresh the sidebar on any changes
-  if (ent[0] == selectedPortal) {
-    log.log('portal guid '+ent[0]+' is the selected portal - re-rendering portal details');
-    renderPortalDetails (selectedPortal);
+  var marker = undefined;
+  if (oldPortal) {
+    // update marker style/highlight and layer
+    marker = window.portals[data.guid];
+    // remove portal from its faction/level specific layer
+    this.removePortalFromMapLayer(marker);
+
+    marker.updateDetails(data);
+
+    window.runHooks('portalAdded', {portal: marker, previousData: previousData});
```

Comment:
the hook is misleading now since the marker is the same marker. Only the data changed.
In this case, we are closer to the `portalDetailLoaded` but with eventually no details (i.e. from core to summary/extended)

Reply:
> the hook is misleading

But that was expected, that's why `previousData` provided. I suppose we should leave it as it.
May be we should add some property to denote placeholders.

But we can revise that from scratch when implementing #217.